src/main/common/parse.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * This file is part of Betaflight.
+ *
+ * Betaflight is free software. You can redistribute this software
+ * and/or modify this software under the terms of the GNU General
+ * Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later
+ * version.
+ *
+ * Betaflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "parse.h"
+
+#include <errno.h>
+#include <stdlib.h>
+
+#include "platform.h"
+
+static intParseResult_t intParseResultOk(long int val, const char * next)
```

Comment:
Or simpy `(intParseResult_t ){.status = INT_PARSE_STATUS_OK, .value = val, .next = next}` literal

Reply:
I like the idea of "constructor" functions because in `C` initializing struct fields is optional. So as a consequence when you add a new field compiler won't warn you. With the function you add a new argument and you will get errors in places where it is used.